templates/test/cluster-template-prow-ha-upgrade.yaml

```diff
@@ -0,0 +1,395 @@
+apiVersion: cluster.x-k8s.io/${UPGRADE_FROM_CAPI_VERSION}
```

Comment:
This template is used by the bootstrap ("management") cluster which is using the latest version, right? So I would suggest changing it so it uses the `CAPI_VERSION` instead and the same for `CAPM3_VERSION`.

Reply:
Here `UPGRADE_FROM_CAPI_VERSION` is v1alpha4 while `CAPI_VERSION` is v1beta1

Reply:
Yes I understand. I'm saying that this template is applied in the initial management cluster, which is running v1beta1. In the cluster that is created we then do `clusterctl init` with v1alpha4, but we never apply this template in there.

Reply:
In fact this template is not used since the v1a5 management cluster does not create any cluster but it should use that template in that case (this will be introduced later as a test workload cluster) v1b1 uses the devenv template

Reply:
Oh I see. In that case I think it would be better to remove it from this PR. We can add it when we actually make use of it.